[Bug report] lang on the 404 page is not set and defaults the Locale to "en"

### Description

The 404 page is not getting normal page variables sent to it. This means the locale in js defaults to "en" when none is present.

![image](https://github.com/vuepress/vuepress-next/assets/3012311/9334a356-6da4-4310-b403-fca182f3d73c)

This is causing js plugins like algolia search to get the wrong lang value. Which should be "en-US" instead of "en".

![image](https://github.com/vuepress/vuepress-next/assets/3012311/a2626b83-e7de-43ae-ba9e-99ab6b89abf7)

Code issue:

```
resolvePageLang: (page: PageData): PageLang => page.lang || 'en',
```

Potential fix:

```
resolvePageLang: (page: PageData): PageLang => page.lang || window.navigator.language || window.navigator.browserLanguage || window.navigator.userLanguage || 'en',
```



### Reproduction

the base example v2.vuepress.vuejs.org/new shows the error when checking the value of 'document.documentElement.lang' from the home page and 404 page.

### Used Package Manager

pnpm

### System Info

```shell
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/zilla-docs - Not found
npm ERR! 404 
npm ERR! 404  'zilla-docs@*' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in: /Users/adanelz/.npm/_logs/2023-06-21T13_52_42_634Z-debug-0.log
```

--- a/packages/client/src/resolvers.ts
+++ b/packages/client/src/resolvers.ts
@@ -92,10 +92,8 @@ export const resolvers = reactive({
    *
    * It would be used as the `lang` attribute of `<html>` tag
    */
-  resolvePageLang: (
-    page: PageData,
-    siteLocale: SiteLocaleData
-  ): PageLang => page.lang || siteLocale.lang || 'en-US',
+  resolvePageLang: (page: PageData, siteLocale: SiteLocaleData): PageLang =>
+    page.lang || siteLocale.lang || 'en-US',
 
   /**
    * Resolve layout component of current page

--- a/packages/client/src/resolvers.ts
+++ b/packages/client/src/resolvers.ts
@@ -92,7 +92,10 @@ export const resolvers = reactive({
    *
    * It would be used as the `lang` attribute of `<html>` tag
    */
-  resolvePageLang: (page: PageData): PageLang => page.lang || 'en',
+  resolvePageLang: (
+    page: PageData,
+    siteLocale: SiteLocaleData
+  ): PageLang => page.lang || siteLocale.lang || 'en-US',
 
   /**
    * Resolve layout component of current page

--- a/packages/client/src/setupGlobalComputed.ts
+++ b/packages/client/src/setupGlobalComputed.ts
@@ -82,7 +82,7 @@ export const setupGlobalComputed = (
       siteLocaleData.value
     )
   )
-  const pageLang = computed(() => 
+  const pageLang = computed(() =>
     resolvers.resolvePageLang(pageData.value, siteLocaleData.value)
   )
   const pageLayout = computed(() =>

--- a/packages/client/src/setupGlobalComputed.ts
+++ b/packages/client/src/setupGlobalComputed.ts
@@ -82,7 +82,9 @@ export const setupGlobalComputed = (
       siteLocaleData.value
     )
   )
-  const pageLang = computed(() => resolvers.resolvePageLang(pageData.value))
+  const pageLang = computed(() => 
+    resolvers.resolvePageLang(pageData.value, siteLocaleData.value)
+  )
   const pageLayout = computed(() =>
     resolvers.resolvePageLayout(pageData.value, layouts.value)
   )

